docs/CNDP_README.md

```diff
@@ -0,0 +1,371 @@
+                        num_q=1
+                        # start queue index
+-                       start_q_idx=22
++                       start_q_idx=6
+                        # RSS using TC filter
+                        setup_tc "${ifaces[$i]}" $num_q $start_q_idx
+                fi
+@@ -218,7 +218,7 @@ fi
+ # Run bessd
+ docker run --name bess -td --restart unless-stopped \
+-       --cpuset-cpus=12-13 \
++       --cpuset-cpus=6-7 \
+        --ulimit memlock=-1 -v /dev/hugepages:/dev/hugepages \
+        -v "$PWD/conf":/opt/bess/bessctl/conf \
+        --net container:pause \
+
+```
+
+7) Modify the script [cndp_reset_upf.sh](../scripts/cndp_reset_upf.sh) to use appropriate PCIe device address, network interface name and set_irq_affinity script in NIC driver.
+
+```
+ACCESS_PCIE=0000:86:00.0
+CORE_PCIE=0000:88:00.0
+ACCESS_IFACE=enp134s0
+CORE_IFACE=enp136s0
+SET_IRQ_AFFINITY=~/nic/driver/ice-1.9.7/scripts/set_irq_affinity
+```
+
+This script is used to stop any running containers, disable irqbalance, set irq affinity to all queues for access and core interface. The script also set XDP socket busy poll settings for access and core interfaces. `set_irq_affinity` script used by this script can be found in the NIC driver install path. irq affinity and AF_XDP busypoll settings are done to get improved network I/O performance. These settings are recommended but not mandatory.
+
+8) From the top level directory call:
+
+```
+$ ./scripts/cndp_reset_upf.sh
+$ ./scripts/docker_setup.sh
+```
+Note: The script [cndp_reset_upf.sh](../scripts/cndp_reset_upf.sh) needs to be executed once before running [docker_setup.sh](../scripts/docker_setup.sh). After that we can execute [docker_setup.sh](../scripts/docker_setup.sh) multiple times if required. The script [cndp_reset_upf.sh](../scripts/cndp_reset_upf.sh) uses dpdk-devbind script from DPDK. To get a copy of it, execute the following command from the UPF's root directory:
+
+```
+$ wget https://raw.githubusercontent.com/DPDK/dpdk/main/usertools/dpdk-devbind.py -O dpdk-devbind.py
+$ chmod +x dpdk-devbind.py
+```
+
+Insert rules into relevant PDR and FAR tables
+
+```
+$ docker exec bess-pfcpiface pfcpiface -config /conf/upf.json -simulate create
+```
+9) From browser, use localhost:8000 to view the UPF pipeline in GUI. If you are remotely connecting to system via ssh, you need to setup a tunnel with local port forwarding.
+
+10) To stop the containers run following command
+
+```
+./scripts/cndp_reset_upf.sh
+```
+
+### Step 4: Run DPDK packet generator in System 2
+
+Build UPF docker image in System 2. Note: If you are behind a proxy make sure to export/setenv http_proxy and https_proxy
+From the top level directory call:
+
+```
+$ make docker-build
+```
+From system 2, bind the two interfaces used by pktgen to DPDK (used to send n/w packets to access/core ). Also setup huge pages in the system.
+
+Modify [pktgen_cndp.bess](../conf/pktgen_cndp.bess) script as follows.
```

Comment:
This name (`pktgen_cndp.bess`) is kind of misleading because the packet generator sends the packets using DPDK.. CNDP is not involved

Reply:
Ok